### PR TITLE
Fix typo in sumKbn, rename to sumKahan

### LINKED ISSUE #14734: Typo in sumKbn (from sums)
Given that [`sumKbn`](https://nim-lang.github.io/Nim/sums.html#sumKbn%2CopenArray%5BT%5D) implements Ka**h**an summation, this is likely a typo and should be `sumKhn`. I don't know if fixing this would be considered a  breaking change, but it is worth mentioning.

EDIT: I assume the solution is to deprecate `sumKbn` and make it an alias for `sumKhn`.

### PR DESCRIPTION
**Pull request by konsumlamm**
Fixes #14734

### CODE REVIEW

File: lib/std/sums.nim [hunk 1]
```diff
@@ -9,7 +9,7 @@
 ## Fast sumation functions.
 
 
-func sumKbn*[T](x: openArray[T]): T =
+func sumKahan*[T](x: openArray[T]): T {.deprecated: [sumKbn: sumKahan].} =
```
Comment (timotheecour) [inline lib/std/sums.nim]: put `{.deprecated: [sumKbn: sumKahan].}` after the proc definition, eg:
```nim
func sumKahan*() =
  impl
{.deprecated: [sumKbn: sumKahan].}
```
(that's why CI fails)

Comment (konsumlamm) [reply lib/std/sums.nim]: Thanks, it should be fixed now.

**Comment by timotheecour** [comment]
see https://github.com/nim-lang/Nim/issues/14734#issuecomment-646904442

**Comment by Araq** [comment]
Needs a changelog entry at least. And probably also a ``.since`` annotation.

**Comment by timotheecour** [comment]
@araq it's contingent on https://github.com/nim-lang/Nim/issues/14734#issuecomment-646904442

**Comment by planetis-m** [comment]
Hey guys there is no typo, the name of the proc is the same as the julia code i ported, (don't have a link at hand right now).

**Comment by timotheecour** [comment]
we know :) see linked discussion

**Comment by alaviss** [comment]
As far as this PR goes, it's clear that this is not an issue that has to be fixed. I think this PR can be closed.

Any other unrelated additions should go into a new PR.

**Comment by konsumlamm** [comment]
You're right.

**Pull request closed without merging**